### ISSUE TITLE
Missing/Unable to access key transcription factor PWMs from CRobustaMotif list

### ISSUE THREAD

**Issue by prakritipaul** [kewiechecki/CrobustaTFs#2]
Hi, 

Firstly, thanks so much for making this resource- it is so helpful! 

I had two comments/questions:
1) Some transcription factors, like Foxg and, are missing from your list (in that it doesn't have these transcription factors as elements), though their information is available on the Aniseed database. 
e.g. for Foxg: https://www.aniseed.cnrs.fr/aniseed/gene/show_selex?unique_id=Cirobu.g00009441

Another example is NEUROG1/2/3 (Neurogenin/C6.129), whose Selex data is not available in Aniseed, but the information for its PWMs from CisBP (in the CisBP/Ciona_intestinalis_2017_11_01_3_41_pm/ directory) appears to be present.

I was wondering how I could access the CisBP information/PWMs of transcription factors directly from the CRobusta list. In layman's terms, how can I do something like "CRobustaMotifs$" and have Foxg/Neurogenin pop up?

I did see that I could contact Matt Weirauch, but thought to post this question here nonetheless. 
 
****
Finally, 2) What do the different versions of Six1/2 mean here? Which PWM should I use, for say, FIMO analysis?
<img width="184" alt="Screen Shot 2021-10-14 at 9 57 22 AM" src="https://user-images.githubusercontent.com/17773458/137332177-c0fb6b42-dcf1-4e95-9d92-c8735b7188e8.png">
****

Thanks so much! 
Prakriti

**Comment by kewiechecki** [comment]
Hi, I apologise for missing this issue.

For FoxG, it seems to be an issue of different gene names being used for the same KHID. To reduce ambiguous gene names, I label motifs by KHID when integrating the data sets, then assign names using data-raw/gene_names.txt. (Selex KHIDs can be seen in data-raw/Selex_seq_Cirobu_All/SELEXgeneID.csv. They can be accessed from the PWMatrixList with ID(CrobustaMotifs). For FoxG, the KHID is KH.C8.774. In the PWMatrixList, this same gene is assigned the name Foxl1. This can be seen with the query
`CrobustaMotifs[[grep("KH.C8.774",ID(CrobustaMotifs))]]`
which should give the output:
```
An object of class PWMatrix
ID: KH2013:KH.C8.774
Name: Foxl1
Matrix Class: Unknown
strand: *
Pseudocounts:
Tags:
$Family_Name
[1] "Forkhead"

$DBID.1
[1] "FoxG"

$Source
[1] "ANISEED"

$strand
[1] "*"

$KHID
[1] "KH2013:KH.C8.774"

$KYID
[1] "KY2019:KY.Chr8.702"

Background:
   A    C    G    T
0.25 0.25 0.25 0.25
Matrix:
     V1    V2    V3        V4    V5    V6        V7    V8    V9   V10   V11
A 0.213 0.218 0.214 0.2057942 0.212 0.225 0.2152152 0.064 0.113 0.061 0.070
C 0.173 0.167 0.174 0.1698302 0.175 0.160 0.1501502 0.057 0.049 0.051 0.039
G 0.301 0.303 0.301 0.2817183 0.303 0.269 0.2902903 0.089 0.744 0.078 0.104
T 0.313 0.312 0.311 0.3426573 0.310 0.346 0.3443443 0.790 0.094 0.810 0.787
    V12   V13   V14   V15       V16       V17   V18   V19       V20   V21
A 0.063 0.106 0.055 0.080 0.2082082 0.2187812 0.192 0.218 0.2202202 0.210
C 0.037 0.040 0.072 0.050 0.1341341 0.1558442 0.187 0.166 0.1691692 0.184
G 0.085 0.774 0.088 0.124 0.2962963 0.3096903 0.264 0.289 0.2862863 0.277
T 0.815 0.080 0.785 0.746 0.3613614 0.3156843 0.357 0.327 0.3243243 0.329
        V22
A 0.2182182
C 0.1741742
G 0.2792793
T 0.3283283
```
For Neurogenin, its absence seems to have been caused by transitioning from Cis-BP 1.02 to Cis-BP 2.00. (The ENSCING00000004039 ID does not have an associated PWM in data-raw/CisBP/Ciona_intestinalis_2020_11_12/)

The numbered variants correspond to motifs for the same TF from different sources. This example script selects a unique PWM for each TF using a combination of filtering by source and by Shannon entropy. Motifs from ANISEED are given higher priority because they've been experimentally validated for _C. robusta_.
```
library(CrobustaTFs)

# extract fields from tags(CrobustaMotifs)
meta <- tfTags(CrobustaMotifs, c('KHID', 'KYID', 'Family_Name', 'Source'))

#select all motifs from ANISEED
selex <- meta[,"Source"]=="ANISEED"
#select motifs from HOMER that are not in ANISEED
homer <- meta[,"Source"]=="HOMER" & !meta[,"KYID"]%in%meta[selex,"KYID"]
#select motifs from CisBP that are not in HOMER or ANISEED
cisbp <- meta[,"Source"]=="CisBP" & !meta[,"KYID"]%in%meta[selex|homer,"KYID"]

#subset the PWMatrixList by the above logical vectors
motifsub <- CrobustaMotifs[selex|homer|cisbp]

#split motifsub into a list of PWMatrixLists by KYID
motifsub <- split(motifsub, tfTags(motifsub,"KYID"))

#select motif with lowest average shannon entropy for each TF
#function that accepts a PWMatrixList and returns the PWMatrix with the lowest Shannon entropy
minent <- function(motifs){
	pwms <- Matrix(motifs)
	#calculates entropy for each position in a PWM and averages them
	en <- sapply(pwms, function(x) mean(apply(x,2,shannon.entropy)))
	return(motifs[which.min(en)])
}

#apply minent to the each TF in motifsub and append the result in a list
motifsub <- Reduce(append, sapply(motifsub, minent))
```

**Comment by prakritipaul** [comment]
Thank you so much- this was very helpful! 
Appreciate it.
